packages/web-api-scan-runner/src/runner/runner.spec.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// tslint:disable: no-object-literal-type-assertion no-unsafe-any
+import 'reflect-metadata';
+
+import { AxeResults } from 'axe-core';
+import { GuidGenerator } from 'common';
+import { Logger } from 'logger';
+import * as MockDate from 'mockdate';
+import { Browser } from 'puppeteer';
+import { AxeScanResults } from 'scanner';
+import { OnDemandPageScanRunResultProvider, PageScanRunReportService } from 'service-library';
+import { ItemType, OnDemandPageScanResult, OnDemandPageScanRunState, ReportFormat, ScanState } from 'storage-documents';
+import { IMock, Mock, Times } from 'typemoq';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+import { ScannerTask } from '../tasks/scanner-task';
+import { WebDriverTask } from '../tasks/web-driver-task';
+import { ScanMetadata } from '../types/scan-metadata';
+import { Runner } from './runner';
+let runner: Runner;
+let browser: Browser;
+let webDriverTaskMock: IMock<WebDriverTask>;
+let onDemandPageScanRunResultProviderMock: IMock<OnDemandPageScanRunResultProvider>;
+let scannerTaskMock: IMock<ScannerTask>;
+let scanMetadataConfig: IMock<ScanMetadataConfig>;
+let loggerMock: IMock<Logger>;
+let pageScanRunReportServiceMock: IMock<PageScanRunReportService>;
+let guidGeneratorMock: IMock<GuidGenerator>;
+
+const scanMetadata: ScanMetadata = {
```

Comment:
you need to rewrite the tests for this:
test cases: 
1.) when axeResult returns error
2.) when axeResult return with violations length  = 0
3.) when axeResult returns with violations length > 0
4.) web driver launch crashes
5.) web driver close crashes
6.) scanner task crashes
7.) cosmos read/write operation crashes